src/apiClients/SendGridApiClient.ts

```diff
@@ -1,14 +1,33 @@
+import { InvalidParametersError } from '../errors';
+import { MailData } from '@sendgrid/helpers/classes/mail';
+import env from '../env';
+import sgMail from '@sendgrid/mail';
+
 class SendGridApiClient {
   private apiKey: string;
 
   constructor(apiKey: string) {
+    if (apiKey === '') {
+      throw new InvalidParametersError('API Key is empty.');
+    }
     this.apiKey = apiKey;
-    // remove this log when this API key is finally used somewhere
-    console.log(`SendGridAPIClient initialized with api key ${this.apiKey}`);
   }
 
-  sendEmail(senderEmailAddress: string, recipientEmailAddress: string, emailBody: string): void {
-    console.log(`sendEmail called with arguments:`, senderEmailAddress, recipientEmailAddress, emailBody);
+  async sendEmail(senderEmailAddress: string, recipientEmailAddress: string, emailBody: string, emailSubject: string) {
+    sgMail.setApiKey(this.apiKey);
+    const mailData: MailData = {
+      to: recipientEmailAddress,
+      from: senderEmailAddress,
+      subject: emailSubject,
+      text: emailBody,
+      mailSettings: {
+        sandboxMode: {
+          enable: env.nodeEnv === 'development',
+        },
+      },
+    };
+    const [sgResponse] = await sgMail.send([mailData]);
```

Comment:
What does the square bracket syntax do on the left hand side?

Reply:
It extracts the first element of the array -- the general name of this type of this syntax is destructuring

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment